src/Validation/ScannableURLsRestController.php

```diff
@@ -189,7 +189,16 @@ public function prepare_item_for_response( $item, $request ) {
 
 			$data = json_decode( $validated_url_post->post_content, true );
 			if ( is_array( $data ) ) {
-				$item['validation_errors'] = wp_list_pluck( $data, 'data' );
+				$item['validation_errors'] = array_map(
+					static function ( $validation_error ) {
+						if ( ! isset( $validation_error['sources'] ) || ! is_array( $validation_error['sources'] ) ) {
+							$validation_error['sources'] = [];
+						}
+
+						return $validation_error;
+					},
+					wp_list_pluck( $data, 'data' )
+				);
```

Comment:
This change isn't needed due to the JS changes, right? Maybe we should leave the PHP changes as-is until we can identify the underlying cause. So only fix the symptoms until we can fix the underlying cause.

So that would mean reverting the PHP changes in this PR but leaving the JS changes.

Reply:
Sounds good, I'll revert the backend changes.

Reply:
Would it be worthwhile to retain the PHP commit and its revert commit so that we can refer back to it when we return to identifying the underlying cause? I'm not bothered by such commits in the history 😄

Reply:
Sure! Since the commits were reverted and then squashed, the history got messed. I've recreated the original commit along with the unit tests update (50e557b) and then reverted it (525b95c). It's now part of the git history.